src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -0,0 +1,219 @@
+            {
+                restApiUri.pop_back();
+            }
+
+            return restApiUri;
+        }
+
+        utility::string_t GetSearchEndpoint(const std::string& restApiUri)
+        {
+            std::string fullSearchAPI = restApiUri;
+            return utility::conversions::to_string_t(fullSearchAPI.append(ManifestSearchPostEndpoint));
+        }
+
+        utility::string_t GetManifestByVersionEndpoint(
+            const std::string& restApiUri, const std::string& packageId, const std::string& version, const std::string& channel)
+        {
+            std::string versionEndpoint = restApiUri;
+            versionEndpoint.append(ManifestByVersionAndChannelGetEndpoint).append(packageId);
+
+            // Add Version Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("?version=").append(version);
+            }
+
+            // Add Channel Query param
+            if (version.empty() && !channel.empty())
+            {
+                versionEndpoint.append("?channel=").append(channel);
+            }
+            else if(!channel.empty())
+            {
+                versionEndpoint.append("&channel=").append(channel);
+            }
+
+            return utility::conversions::to_string_t(versionEndpoint);
+        }
+
+        std::optional<std::string> GetStringFromJsonStringValue(const web::json::value& value)
+        {
+            if (value.is_null() || !value.is_string())
+            {
+                return {};
+            }
+
+            return utility::conversions::to_utf8string(value.as_string());
+        }
+
+        std::string ValidateAndGetStringFromJsonStringValue(const web::json::value& value)
+        {
+            std::optional<std::string> result = GetStringFromJsonStringValue(value);
+
+            if (!result.has_value() && Utility::IsEmptyOrWhitespace(result.value()))
+            {
+                THROW_HR_MSG(E_UNEXPECTED, "Missing required value");
+            }
+
+            return result.value();
+        }
+    }
+
+    Interface::Interface(const std::string& restApi)
+    {
+         m_restApiUri = GetRestAPIBaseUri(restApi);
+         m_searchEndpoint = GetSearchEndpoint(m_restApiUri);
+    }
+
+    IRestClient::SearchResult Interface::Search(const SearchRequest& request) const
+    {
+        SearchResult result;
+
+        try
+        {
+            // TODO: Handle continuation token.
+            HttpClientHelper clientHelper{ m_searchEndpoint };
+            web::json::value jsonObject = clientHelper.HandlePost(GetSearchBody(request));
+
+            // Parse json and add results to SearchResult.
+            if (jsonObject.is_null())
+            {
+                return result;
+            }
+
+            auto& dataArray = jsonObject.at(GetJsonKeyNameString(Data)).as_array();
+
+            for (auto& manifestItem : dataArray)
+            {
+                try

```

Comment:
>try [](start = 16, length = 3)

Exceptions are not for control flow.  We should try to avoid them when we can write logic to better handle the situation.  Certainly we do not want to throw in the underlying function call just to catch it here.